src/moveit_visual_tools.cpp

```diff
@@ -1395,26 +1395,34 @@ bool MoveItVisualTools::publishRobotState(const std::vector<double> joint_positi
 }
 
 bool MoveItVisualTools::publishRobotState(const robot_state::RobotStatePtr& robot_state,
-                                          const rviz_visual_tools::colors& color)
+                                          const rviz_visual_tools::colors& color,
+                                          const std::vector<std::string>& highlight_links)
 {
-  return publishRobotState(*robot_state.get(), color);
+  return publishRobotState(*robot_state.get(), color, highlight_links);
 }
 
 bool MoveItVisualTools::publishRobotState(const robot_state::RobotState& robot_state,
-                                          const rviz_visual_tools::colors& color)
+                                          const rviz_visual_tools::colors& color,
+                                          const std::vector<std::string>& highlight_links)
 {
+  // when only a subset of links should be colored, the default message is used
+  rviz_visual_tools::colors base_color = color;
+  if (!highlight_links.empty())
+    base_color = rviz_visual_tools::DEFAULT;
+
   // Reference to the correctly colored version of message (they are cached)
   // May not exist yet but this will create it
-  moveit_msgs::DisplayRobotState& display_robot_msg = display_robot_msgs_[color];
+  moveit_msgs::DisplayRobotState& display_robot_msg = display_robot_msgs_[base_color];
 
   // Check if a robot state message already exists for this color
   if (display_robot_msg.highlight_links.size() == 0)  // has not been colored yet, lets create that
   {
     if (color != rviz_visual_tools::DEFAULT)  // ignore color highlights when set to default
     {
       // Get links names
-      const std::vector<const moveit::core::LinkModel*>& link_names =
-          robot_state.getRobotModel()->getLinkModelsWithCollisionGeometry();
+      const std::vector<std::string>& link_names =
+          highlight_links.empty() ? robot_state.getRobotModel()->getLinkModelNamesWithCollisionGeometry() :
```

Comment:
getLinkModelNamesWithCollisionGeometry() may limit you... there is visual geometry that you might want to highlight that isn't listed in collision geometry. can we make this more generic?

Reply:
ah i realize this was my previous behavior. now im wondering if it makes sense, but its a bit off topic i guess

Reply:
I think using this function is perfectly fine. Links where there is only a visual defined use the underlying geometry as collision shape. I tested this and the function really highlights all links even though there are only visual shapes defined in the urdf.